app/Main.hs

```diff
@@ -1,10 +1,86 @@
 -- SPDX-FileCopyrightText: 2020 TBD
 --
 -- SPDX-License-Identifier: LicenseRef-Proprietary
-module Main (main) where
+{-# LANGUAGE ApplicativeDo #-}
+{-# OPTIONS_GHC -Wno-orphans #-}
 
-someFunc :: IO ()
-someFunc = putTextLn "someFunc"
+module Main
+  ( main
+  ) where
+
+import qualified Data.Map as Map
+import Data.Version (showVersion)
+import Main.Utf8 (withUtf8)
+import qualified Options.Applicative as Opt
+import Options.Applicative.Help.Pretty (Doc, linebreak)
+import Paths_globacap (version)
+
+import qualified Lorentz as L
+import Lorentz (mt)
+import Lorentz.ContractRegistry
+import Michelson.Typed (starNotes)
+import Morley.CLI (addressOption, mTextOption)
+import Tezos.Address (Address)
+import Util.CLI (mkCLOptionParser)
+import Util.Named
+
+import Indigo.Contracts.Holdings (holdingsContract)
+import qualified Indigo.Contracts.Holdings as H
+
+programInfo :: L.DGitRevision -> Opt.ParserInfo CmdLnArgs
+programInfo gitRev = Opt.info (Opt.helper <*> versionOption <*> argParser contracts gitRev) $
+  mconcat
+  [ Opt.fullDesc
+  , Opt.progDesc "Tezos Globacap contracts registry"
+  , Opt.header "Contracts for Tezos Globacap project"
+  , Opt.footerDoc usageDoc
+  ]
+  where
+    versionOption = Opt.infoOption ("globacap-" <> showVersion version)
+      (Opt.long "version" <> Opt.help "Show version.")
+
+usageDoc :: Maybe Doc
+usageDoc = Just $ mconcat
+   [ "You can use help for specific COMMAND", linebreak
+   , "EXAMPLE:", linebreak
+   , "  globacap print --help", linebreak
+   ]
+
+contracts :: ContractRegistry
+contracts = ContractRegistry $ Map.fromList
+  [ "Holdings" ?:: ContractInfo
+    { ciContract = holdingsContract
+    , ciIsDocumented = True
+    , ciStorageParser = Just holdingsStorageParser
+    , ciCompilationOptions = L.defaultCompilationOptions
+    , ciStorageNotes = starNotes
```

Comment:
I am afraid people won't like the lack of annotations in the storage. Hopefully https://gitlab.com/morley-framework/morley/-/merge_requests/185 will be merged soon, but for now let's add them manually.

Reply:
:pepe_sad:

Reply:
```
storage (pair (big_map %ledger address
                               (pair (nat %balance)
                                     (map %approvals address
                                                     nat)))
              (pair (pair (pair (pair %tokenMeta (string %name)
                                                 (pair (string %symbol)
                                                       (string %id)))
                                (option %mbSafelistAddress address))
                          (pair (address %owner)
                                (address %admin)))
                    (pair (pair (option %mbNewAdmin address)
                                (bool %paused))
                          (pair (bool %transferable)
                                (nat %totalSupply)))));
```